oauth-common/src/main/java/io/strimzi/kafka/oauth/common/OAuthAuthenticator.java

```diff
@@ -167,6 +167,90 @@ public static TokenInfo loginWithClientSecret(URI tokenEndpointUrl, SSLSocketFac
+     * @return A TokenInfo with access token and information extracted from it
+     * @throws IOException If the request to the authorization server has failed
+     * @throws IllegalStateException If the response from the authorization server could not be handled
+     */
+    @SuppressWarnings("checkstyle:ParameterNumber")
+    public static TokenInfo loginWithClientAssertion(URI tokenEndpointUrl, SSLSocketFactory socketFactory,
+                                                     HostnameVerifier hostnameVerifier,
+                                                     String clientId, String clientAssertion, String clientAssertionType, boolean isJwt,
+                                                     PrincipalExtractor principalExtractor, String scope, String audience, boolean includeAcceptHeader) throws IOException {
+
+        return loginWithClientAssertion(tokenEndpointUrl, socketFactory, hostnameVerifier,
+                clientId, clientAssertion, clientAssertionType, isJwt, principalExtractor, scope, audience, HttpUtil.DEFAULT_CONNECT_TIMEOUT, HttpUtil.DEFAULT_READ_TIMEOUT, null, 0, 0, includeAcceptHeader);
+    }
+
+    /**
+     * Obtain an access token wrapped into TokenInfo by authenticating to the authorization server's token endpoint
+     * using client_credentials grant with client assertion (clientId + assertion + assertion type).
+     *
+     * @param tokenEndpointUrl A token endpoint url
+     * @param socketFactory A socket factory to use with 'https'
+     * @param hostnameVerifier A hostname verifier to use with 'https'
+     * @param clientId A client id
+     * @param clientAssertion A client assertion
+     * @param clientAssertionType A client assertion type
+     * @param isJwt If the returned token is expected to be a JWT token
+     * @param principalExtractor A PrincipalExtractor to use to determine the principal (user id)
+     * @param scope A scope to request when authenticating
+     * @param audience An 'audience' attribute to set on the request when authenticating
+     * @param connectTimeout A connect timeout in seconds
+     * @param readTimeout A read timeout in seconds
+     * @param metrics A MetricsHandler object to receive metrics collection callbacks
+     * @param retries A maximum number of retries if the request fails due to network, or unexpected response status
+     * @param retryPauseMillis A pause between consecutive requests
+     * @param includeAcceptHeader Should we skip sending the Accept header when making outbound http requests
+     * @return A TokenInfo with access token and information extracted from it
+     * @throws IOException If the request to the authorization server has failed
+     * @throws IllegalStateException If the response from the authorization server could not be handled
+     */
+    @SuppressWarnings("checkstyle:ParameterNumber")
+    public static TokenInfo loginWithClientAssertion(URI tokenEndpointUrl, SSLSocketFactory socketFactory,
```

Comment:
Can we put each parameter on a separate line.

Reply:
I aded logic that if both are set, the location (file) is used rather than the token, and warning is logged.